src/db/bookmark.ts

```diff
@@ -8,6 +8,9 @@ export const getBookmarkDataWithContent = async (): Promise<
 > => {
   const session = await getServerSession(authOptions);
   const userId = session?.user.id;
+  if (!userId) {
```

Comment:
this isn't required as the layout will navigate the user before even this function gets invoked

Reply:
@ItsFlash10  server function are http under the hood. its good pratice to make sure that api is also protected.

Reply:
Yes I agree with @paahaad here, server actions will get converted into sort of `grpc` like calls